test/es-module/test-esm-assertionless-javascript-import.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
```

Comment:
Can you duplicate this test cases into an ES module please? Dynamic imports in CJS and in ESM use different code paths.

Reply:
Sure, should we do the same for `test-esm-assertionless-json-import.js`? I don't see an ESM version of that one.

Reply:
Oh that's an oversight of mine. It doesn't really belong to this PR, and I'm not sure it deserves its own PR (it being non-standard and all). I'd say we're good for now.